### ISSUE TITLE
Proposal for bringing clarity to the Notary Project branding

### ISSUE THREAD

**Issue by toddysm** [notaryproject/.github#35]
@notaryproject/notaryproject-governance-maintainers In the last Notary Project meeting ([notes](https://hackmd.io/_vrqBGAOSUC_VWvFzWruZw?both#Jun-15-2023) and [meeting recording](https://www.youtube.com/watch?v=x9M9xaryV_M)) we discussed how to clarify the branding of the Notary Project and make it clear to people who are interested to use the tools and participate in the community what means what. This is related also to the following two items https://github.com/notaryproject/notaryproject/pull/262 and https://github.com/notaryproject/.github/pull/32. 

In the meeting, we discussed the following changes, which I am posting here for discussion by maintainers.
1. Archive the `notaryproject` repository and create a new repository called `specifications` that will contain the specifications that are across any tools as well as used by any external tools or projects that want to interoperate with our tools. 
    1.1 Inside the new repository, we will structure the specifications in folders to provide more clarity. Right now, we have two main specification areas - one for signatures and signing workflow for OCI artifacts, and one for Notation plugin implementation. Those should be in separate folders. In the future, we will have more specifications that will be placed in their own folders. Each folder will have a README.md to explain the purpose of each document inside the folder. 
    1.2 The repository will have its own README.md to explain the purpose of the repository as well as link to the READMEs in each individual folder.
    1.3 Current `notaryproject` repository contains folders for `requirements` and `scenarios`. Requirements can be captured as GitHub issues in the future and scenarios can be documented in the GitHub issues, HackMD documents linked from the GitHub issues or on the project's website in `notaryproject.dev` repository. Hence, the proposal is to not move those to the new repository.
    1.4 Current `notaryproject` repository contains the latest security reports. The latest security reports cover multiple repositories under the organization. _Note:_ there are also some security reports in the `notary` repository that are specific to the implementation in that repository. The proposal is to migrate the latest security reports and the threat model to the `.github` repository and create a relevant directory structure in it. Also, to link to the security reports from the READMEs of the relevant repositories they cover.
2. We need to update each individual repository README.md with the following information: link to the overview, purpose of the repository, is it in active development or not, is it archived or not.
3. We need to have a process for archiving repositories under the organization and make sure that we clean up all repositories before the release.

We also discussed to acknowledge the terminology we use and how should we use it. The confusion comes from the use of terms like _"Notary"_, _"Notary V2"_, _"Notary Project"_, _"Notation"_. Here is the proposed (very draft) language:

1. The name of the GitHub organization is _"Notary Project"_. When used this term has an all-encompassing meaning and will refer to the GitHub organization, the community, all specifications, and all the repositories under the organization including tools that are released by the community. Because this term is too broad, it is recommended to always use it with additional clarifiers unless we mean all of the above. For example, _"Notary Project's signature specification"_ or _"Notarty Project's signing workflow"_ or _"Notary Project's Notation tool"_ (see about the last one below). The term should be used as an encompassing brand.
2. The name _"Notary"_ refers to the TUF-based implementation from the [`notary`](https://github.com/notaryproject/notary) repository. This is the only meaning of that term and we should use it only if we mean the TUF-based implementation.
3. The name _"Notary V2"_ or _"Notary v2"_ has no corresponding implementation or any relevant assets under the organization. Thus, this name is unclear and has no clear meaning for anything that _Notary Project community_ collaborates on. Hence, this name should not be used in any context relevant to our work.
4. The name _"Notation"_ refers to the implementation of the CLI and the libraries in the `notation-go` and `notation-core-go` repositories. If not clarified, the term _"Notation"_ will mean the CLI. Of course, the CLI meaning can be clarified with _"Notation CLI"_. If we want to address the libraries, we should always clarify with the corresponding library in mind. For example, _"Notation Go"_, respectively _"Notation Go library"_ for the library in the `notation-go` repository as well as _"Notation Core Go"_, respectively _"Notation Core Go library"_ for the library in the `notation-core-go` repository.
5. The term _"Notary Project release"_ is also confusing because it doesn't clarify what is released (see above for the meaning of the term). We should use specific clarifiers for the releases. For example _"Notary Project's signature specification release"_ or _"Notary Project's Notation CLI release"_ or simply _"Notation CLI release"_.
6. Last but not least, we would like to propose a change in the Notary Project logo to include the word "Project".

Note that the release of the specs and the Notation CLI are contingent on the branding changes that we describe above. I am appealing to the maintainers to take an active role in discussing and agreeing on the changes as soon as possible.

CC:// @TheFoxAtWork and @mattfarina for your feedback and comments on the above.

**Comment by toddysm** [comment]
One additional thing that was brought to my attention - the concepts of [sub-project](https://github.com/notaryproject/.github/issues/10) and [working-group](https://github.com/notaryproject/.github/issues/9) would be very helpful in clarifying the use of the repositories and terminology. _Notary_, _Notation_, _Notation libraries_ are good candidates for _"sub-projects"_ of _Notary Project_.

**Comment by toddysm** [comment]
Updated the issue to use numbered list as per suggestion from @iamsamirzon on the Notary Project community call.

**Comment by sajayantony** [comment]
SGTM and thank you @toddysm +1 on getting the terminologies sorted out. As per the call my understanding is that we would prioritize the terminology PR so that we can document these in #32. 
Recommend other @notaryproject/notaryproject-org-maintainers  to please comment with an LGTM or comment on this issue if there is any concern on this.

**Comment by yizha1** [comment]
LGTM

Thanks you @toddysm for the proposals. I would like to work with other maintainers of each repository to update the repo readme.md file based on this issue and https://github.com/notaryproject/.github/pull/32, and review current governance documents on archiving repositories.

**Comment by Two-Hearts** [comment]
LGTM. Thanks @toddysm for the proposal!

**Comment by Miranlfk** [comment]
LGTM. 

Would appreciate to see the terminology and branding improvements suggested by @toddysm to reduce/remove any confusion for users and community members!

**Comment by FeynmanZhou** [comment]
@toddysm  Thanks for putting things together. The proposal of clarifying project terminology and archiving/restructuring the `notaryproject` repository is reasonable and doable. It's worth documenting these terminologies on https://notaryproject.dev/. I am happy to help.

**Comment by FeynmanZhou** [comment]
Given that we are close to the Notation v1.0.0 release, I would suggest fixing the naming/terminology issue before Notation v1.0.0. IMO, archiving/refactoring the `notaryproject` repository will not be a blocking issue for releasing Notation v1.0.0 and it could be marked as the first priority after v1.0.0. We could work in parallel to document the archiving process mentioned in #36 , archive `notaryproject` repo, and create a new `specifications`  repo.

**Comment by shizhMSFT** [comment]
LGTM and thank @toddysm for putting all the things together.

**Comment by TheFoxAtWork** [comment]
> 1.3 Current notaryproject repository contains folders for requirements and scenarios. Requirements can be captured as GitHub issues in the future and scenarios can be documented in the GitHub issues, HackMD documents linked from the GitHub issues or on the project's website in notaryproject.dev repository. Hence, the proposal is to not move those to the new repository.

Requirements may be captured as GH issues however it may also be prudent to record high level 'requirements' or design principles as Scoping Goals/ Guide that the project and its community members may continuously refer back to when determining overall features, design criteria, and the acceptable characteristics for PRs.

> We need to update each individual repository README.md with the following information: link to the overview, purpose of the repository, is it in active development or not, is it archived or not

[Falco has a fantastic example of how they captured this](https://github.com/falcosecurity/evolution/blob/main/README.md#core), I highly recommend taking a look and adopting a version of it.

> We need to have a process for archiving repositories under the organization and make sure that we clean up all repositories before the release.

Recommend this be included as a dedicated governance file for Archiving. Each repo should link back to that governance file. It should also include how the determination for Archive occurs and detail how Archival (once determined) is executed. 

> "Notary Project" ... The term should be used as an encompassing brand.

Wonderful, please make sure this is apparent on the documentation with an initial link back to the definition of what that encompasses the first time the phrase is introduced in repos.

> prioritize the terminology PR so that we can document these in https://github.com/notaryproject/.github/pull/32.

Agreed. The sooner terminology can be updated, the faster the project may move forward.  I've made comments on PR 32 prior to reviewing this issue. **Please consider those comments superseded by the decisions from this issue as appropriate.**

Thank you all so much for taking the time to record this discussion's outcomes and recommendations. This will greatly assist new contributors and potential adopters at understanding this Project's varied repositories, goals, and objectives.

**Comment by FeynmanZhou** [comment]
> Recommend this be included as a dedicated governance file for Archiving. Each repo should link back to that governance file. It should also include how the determination for Archive occurs and detail how Archival (once determined) is executed.

Thanks @TheFoxAtWork . I would like to create a dedicated file to clarify the archiving process in this repo.

**Comment by FeynmanZhou** [comment]
link #37

**Comment by NiazFK** [comment]
IMO the terms _"Notary Project's signature specification"_ or _"Notary Project's signing workflow"_ or _"Notary Project's Notation tool"_ are overly complicated and confusing, specially for end users. Signature specification and workflows are specific to the TUF (Notary) and non-TUF implementations (NotaryV2 which was renamed to Notation). The newer signature specification should therefore be called Notation signature/signature spec.
To keep things simple and clearly delineated, I'd propose two set of tools/specifications under "Notary Project" GitHub organization.
1. Notary - the TUF based container image signing solution
2. Notation - the newer non-TUF based image signing solution

If we wish to retain [notaryproject ](https://github.com/notaryproject/notaryproject) repo, I'd move all newer requirements and specs under a `Notation` subfolder.

**Comment by toddysm** [comment]
@NiazFK and @gokarnm, there were several discussions in Notary Project meetings and in various issues. To summarize the latest:
- Update all references of "Notary", "Notary v2" to "Notary Project" when the overall effort is meant. If particular tools like the `notary` CLI is meant to use the appropriate casing and clarify that we mean the CLI tool. Related work items: 
    - https://github.com/notaryproject/notaryproject/issues/244
    - https://github.com/notaryproject/roadmap/issues/89
    - https://github.com/notaryproject/meeting-notes/issues/15
- For the benefits of the implementers: Update the overview of the Notary Project to clearly describe the goals of the project, the evolution, and how the TUF-based implementation in `notary` repository differs from the signature specification and the `notation` implementation; document what requirements and scenarios the signature specification enables that are not enabled by the TUF-based implementation (this should be done in https://github.com/notaryproject/.github/pull/32). Other related work items:
    - https://github.com/notaryproject/notaryproject/issues/259
- Update the READMEs in each repository to describe the purpose of the repository only and point to the overview in the `.github` repository (the one discussed in https://github.com/notaryproject/.github/pull/32). Work items:
    - https://github.com/notaryproject/notaryproject/issues/260
    - https://github.com/notaryproject/notation/issues/715
    - https://github.com/notaryproject/notation-go/issues/323
    - https://github.com/notaryproject/notary/pull/1685
    - https://github.com/notaryproject/notaryproject.dev/issues/288
    - https://github.com/notaryproject/notation-core-go/issues/150
    - https://github.com/notaryproject/notation-hashicorp-vault/issues/5
    - https://github.com/notaryproject/tuf/issues/43
    - https://github.com/notaryproject/meeting-notes/issues/17
    - https://github.com/notaryproject/roadmap/issues/91
    - For `notation-action` TBD
- For the benefits of consumers: Create a FAQ on the documentation site that describe the above in user-friendly language (work item TBD)
- Rename `notaryproject/notaryproject` repository to `notaryproject/specifications`. Related work item 
    - #38
- Rename `notaryproject/notary` repository to `notaryproject/notary-tuf` (work item TBD)
- Retire `notaryproject/tuf` repository (work item TBD)

All of the above to be completed before the announcement of the release of the Notary Project specification and the Notation CLI and library stable release.

Let's use this issue as a discussion of the plan. 

@NiazFK, @gokarnm, @yizha1, @FeynmanZhou, @SteveLasker, @justincormack, @vaninrao10, @priteshbandi  - as maintainers, I am looking for super majority acceptance (LGTM in a comment) of the above plan. As the one proposing this plan, my vote is +1.

CC:// @iamsamirzon

**Comment by gokarnm** [comment]
LGTM! Thanks for putting the updated plan with all the tracking work items!

**Comment by yizha1** [comment]
LGTM. Thanks Toddy. IMO, as long as we kick off retiring `notaryproject/tuf` repository, it won't block the release.

**Comment by FeynmanZhou** [comment]
LGTM.

**Comment by FeynmanZhou** [comment]
@toddysm I created another issue to track the redesign of the project logo to `Notary Project` https://github.com/notaryproject/.github/issues/43

**Comment by NiazFK** [comment]
LGTM

**Comment by iamsamirzon** [comment]
LGTM but IAMNAM

**Comment by FeynmanZhou** [comment]
> For the benefits of consumers: Create a FAQ on the documentation site that describe the above in user-friendly language (work item TBD)

@toddysm Do you mean to clarify terminologies like "Notary", "Notary V2", "Notary Project", "Notation" on the Notary Project documentation?

**Comment by toddysm** [comment]
What I mean is to create a FAQ that describes what is the difference between `notary`, TUF, Notary Project specification, and `notation`. What is what, what it means, how it works, and how it is different from the others? Hope this makes sense.

**Comment by priteshbandi** [comment]
LGTM

**Comment by yizha1** [comment]
Thanks for the votes. `6` out of `9` maintainers approved this [proposal](https://github.com/notaryproject/.github/issues/35#issuecomment-1642938826), we have reached a two-thirds supermajority. The next step is to fix the issues listed in the proposals. As discussed in 07/24 [community call](https://hackmd.io/_vrqBGAOSUC_VWvFzWruZw?both#Meeting-Notes), the plan is to send out relevant PRs not later than Jul 27 PDT time, we will review the status in Jul 27 PDT community call. I will also take an action to update the existing governance guide to make it clear that super majority means a two-thirds supermajority, which is also required in [graduation_criteria](https://github.com/cncf/toc/blob/main/process/graduation_criteria.md).

**Comment by SteveLasker** [comment]
LGTM

**Comment by yizha1** [comment]
As follow-up on the [proposal](https://github.com/notaryproject/.github/issues/35#issuecomment-1642938826), here are the PRs and issues for review:

Repo Renaming (**Requesting 2/3 supermajority of org and gov maintainers**)
- https://github.com/notaryproject/.github/issues/38
- https://github.com/notaryproject/.github/issues/44

Fix naming (**requesting repo maintainers**)

- https://github.com/notaryproject/notaryproject/pull/256
- https://github.com/notaryproject/notaryproject/pull/262
- https://github.com/notaryproject/meeting-notes/pull/16

Updating Readme.md (**requesting repo maintainers**)

- https://github.com/notaryproject/.github/pull/32
- https://github.com/notaryproject/notation/pull/750
- https://github.com/notaryproject/notation-go/pull/343
- https://github.com/notaryproject/notation-core-go/pull/158
- https://github.com/notaryproject/notary/pull/1685
- https://github.com/notaryproject/notation-action/pull/18
- https://github.com/notaryproject/notaryproject.dev/pull/295
- https://github.com/notaryproject/notaryproject/pull/263
- https://github.com/notaryproject/notation-hashicorp-vault/pull/9
- https://github.com/notaryproject/tuf/pull/45
- https://github.com/notaryproject/meeting-notes/pull/18
- https://github.com/notaryproject/roadmap/pull/92

Archiving (**Requesting 2/3 supermajority of org and gov maintainers**)
- https://github.com/notaryproject/.github/pull/37
- https://github.com/notaryproject/.github/issues/47 (Depending on Archiving process)

Website: (**requesting repo maintainers**)
- https://github.com/notaryproject/notaryproject.dev/pull/326
- https://github.com/notaryproject/notaryproject.dev/pull/327

/cc: @toddysm @justincormack @gokarnm @priteshbandi @iamsamirzon @FeynmanZhou @SteveLasker @vaninrao10

**Comment by yizha1** [comment]
@iamsamirzon We didn't discuss your question during community meeting on Jul 31. I think we should use `the Notary Project`, not `The Notary Project`, except it is used at the beginning of the Sentence. /cc @toddysm any comments?

**Comment by toddysm** [comment]
@yizha1 This was discussed in the meeting on 8/3 and the agreement is to use lowercase "the"

**Comment by yizha1** [comment]
Per the discussion in the meeting on 8/3, [PR for adding glossary](https://github.com/notaryproject/notaryproject.dev/pull/326) is not blocking for release.

**Comment by yizha1** [comment]
We completed major work items per this [proposal](https://github.com/notaryproject/.github/issues/35#issuecomment-1642938826). However, there are still some work items to be closed. Most of them are related to update `README.md` for each repo. Here is the prioritized list for review:

`README.md` for `notation-go` and `notation-core-go` repo:

- https://github.com/notaryproject/notation-go/pull/343
- https://github.com/notaryproject/notation-core-go/pull/158

Archiving process:
- https://github.com/notaryproject/.github/pull/37

`README.md` for other repos:

- https://github.com/notaryproject/tuf/pull/45
- https://github.com/notaryproject/meeting-notes/pull/18
- https://github.com/notaryproject/roadmap/pull/92
- https://github.com/notaryproject/notation-action/pull/18
- https://github.com/notaryproject/notaryproject.dev/pull/295

**Comment by yizha1** [comment]
Closed this issue as most of the work items defined [here](https://github.com/notaryproject/.github/issues/35#issuecomment-1653281321) are completed. The only work item left is tracked by this issue https://github.com/notaryproject/.github/issues/47
